### DISCUSSION TITLE
Generating a timestamp without 8 tuples

### DISCUSSION THREAD

**Discussion by nikrawlinson** [micropython/micropython#10125] [Using MicroPython]
I am using an API to access data that provides times and dates for events in the following format:

YYYY-MM-DDTHH:MM:SS

So, for example:

2202-11-29T02:29:30

I can convert this to a timestamp in Python so that I can calculate seconds since epoch time, but am having difficulty doing the same in Micropython using time or utime as I don't have eight tuples. Can anyone suggest how I could achieve this?

Thanks

**Comment by fivdi** [comment]
I think you can achieve what you would like to achieve by creating an 8-tuple that looks looks like this:
```python
(YYYY, MM, DD, HH, MM, SS, 0, 0)
```
The last two entries in the tuple which are week-day and year-day can be set to 0.

```
>>> 
MicroPython v1.19.1 on 2022-06-18; ESP32S3 module with ESP32S3
Type "help()" for more information.
>>> import time
>>> t=(2022, 11, 29, 2, 29, 30, 0, 0)
>>> time.mktime(t)
723004170
>>> 
```
In most MicroPython ports (all?) `time.mktime()` only takes the first 6 entries in the 8-tuple into account. For example, for the ESP32 see [here](https://github.com/micropython/micropython/blob/v1.19.1/ports/esp32/modutime.c#L62-L75).

**Reply by nikrawlinson** [reply]
So simple. Thanks very much.
